File: level01/repeat_alpha/repeat_alpha.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   repeat_alpha.c                                     :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: whumfrey <marvin@42.fr>                    +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2018/11/25 17:56:20 by whumfrey          #+#    #+#             */
/*   Updated: 2018/12/10 16:32:31 by whumfrey         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <unistd.h>

void	ft_repeat(char *str)
{
	int	i;
	int	j;
	
	i = 0;
	while (str[i])
	{
		if (str[i] >= 'a' && str[i] <= 'z')
		{
			j = str[i] - 'a';
			while(j--)
				write(1, &str[i], 1);
		}
		if (str[i] >= 'A' && str[i] <= 'Z')
		{
			j = str[i] - 'A';
			while(j--)
				write(1, &str[i], 1);
		}
		write(1, &str[i], 1);
		i++;
	}
}

int		main(int argc, char **argv)
{
	if (argc == 2)
	{
		ft_repeat(argv[1]);
	}
	write(1, "\n", 1);
}

File: level01/repeat_alpha/repeat_alpha.py
def ft_repeat(str):
	i = 0
	for x in str:
		j = 0
		if (ord(x) >= ord('a') and ord(x) <= ord('z')):
			j = ord(x) - ord('a') + 1
			while j:
				print (x, end = "")
				j -= 1
		elif (ord(x) >= ord('A') and ord(x) <= ord('Z')):
			j = ord(x) - ord('A') + 1
			while j:
				print (x, end = "")
				j -= 1
		else:
			print (x, end = "")

import sys
s = sys.argv
s = s[1 : ]
if (len(s) == 1):
	str = "".join (s)
	ft_repeat(str)
print("")